### PR TITLE
added json mirror - flexible client and server bound object

### PR DESCRIPTION
**Pull request by dioptre**

### CODE REVIEW

**Comment by dioptre** [comment]
added raw-html too:
vbox(
	naked("""<script>alert('hixesdz');</script>""")
)

**Comment by dioptre** [comment]
icons were broken for me and had to move around dependencies as per http://stackoverflow.com/questions/30608194/polymer-1-0-default-icon-set-in-iron-icons-not-working-using-blaze-meteor-temp

**Comment by dioptre** [comment]
Hey guys whats the status with this? I've got other people wanting to use it :)

**Comment by rohitvarkey** [comment]
@dioptre As @shashi mentioned in the previous PR, please make one PR for one feature/fix. This PR does 3 things at once and makes changes in 95 files (even though a good fraction is font file additions), making it hard to review correctly. It would be great if you could separate these into different PR's. 

Regarding `naked`, You could also do `Elem(:h1, "gggg")`  instead of `naked(...)`. Using `Elem` would let you use Patchworks diffing features if you want to update things dynamically. `naked` will also get transformed to an `Elem` at the end but if you have a bunch of html tags there instead of just one you'd end up having to update the html string, while Patchwork could update only the attributes that changed. But I guess it is useful when you just want to output HTML you already have somewhere and not bother changing them to `Elem's`.

File: src/library/naked.jl [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export naked
+
+@api naked => (Naked <: Tile) begin
```
Comment (shashi) [inline src/library/naked.jl]: Can we just call this `html`?

File: src/cli/serve.jl [hunk 2]
```diff
@@ -160,9 +160,8 @@ uisocket(dir) = (req) -> begin
 
     foreach(asset -> write(sock, JSON.json(import_cmd(asset))),
          window.assets)
-
     main = loadfile(file)
-
+
```
Comment (shashi) [inline src/cli/serve.jl]: should remove these changes

File: examples/slides.jl [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+            - Images
+            - SymPy
+            """ |> lineheight(2em) |> Escher.fontsize(2em)
+        ),
+        vbox(
+            title(3, "Thank you for listening!"), vskip(1em),
+            title(2, "https://shashi.github.io/Escher.jl"), vskip(1em),
+            title(1, "or just google Escher.jl"),
+        ),
+    ])
```
Comment (shashi) [inline examples/slides.jl]: I'll put this in after some updates, don't include it in for now.

File: examples/repl.jl [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    vbox(
+        vcat(
+            repl.tiles.tiles,
+            vbox(
+                code_io(code, cell_sig, linenumbers=false, name=:code),
+                lift(showoutput, cell_sig)
+            )
+        )
+    )
+end
```
Comment (shashi) [inline examples/repl.jl]: this as well, I'll make the changes here and put this in. This code is obsolete and I'd very much like to use the Components stuff here

File: assets/bower_components/iron-icons/iron-icons.html [hunk 5]
```diff
@@ -301,3 +300,4 @@
 <g id="zoom-out"><path d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14zM7 9h5v1H7z"/></g>
 </defs></svg>
 </iron-iconset-svg>
+<link rel="import" href="../iron-iconset-svg/iron-iconset-svg.html">
```
Comment (shashi) [inline assets/bower_components/iron-icons/iron-icons.html]: hmm,  does this fix anything?

File: src/library/naked.jl (hunk 1 above)
Comment (shashi) [inline src/library/naked.jl]: or `raw` or `rawhtml`

File: assets/bower_components/iron-icons/iron-icons.html (hunk 5 above)
Comment (dioptre) [inline assets/bower_components/iron-icons/iron-icons.html]: yes - it fixes the note http://stackoverflow.com/questions/30608194/polymer-1-0-default-icon-set-in-iron-icons-not-working-using-blaze-meteor-temp

File: src/library/naked.jl (hunk 1 above)
Comment (dioptre) [inline src/library/naked.jl]: I wanted to make it unique for the future as raw/html/etc are very generic I thought there'd be less chance of a conflict.

**Comment by dioptre** [comment]
@rohitvarkey Elem(:h1, "gggg") won't let you put in script tags, javascript, metatags, etc I don't think.

I'm new to Julia, github etc - so I don't get why it just automatically appended the PR to my original one.

**Comment by rohitvarkey** [comment]
@dioptre Just tried `Elem(:script, """alert("hello")""")` and it worked. Anything you could represent in the DOM should be representable as Elems too, I think! `naked` is a more convenient way to express it though.

Github makes Pull Requests based on the branch you are making it from. As all your commits are on the `master` branch, all the changes have appeared as in one PR. The workflow I follow usually is to:

1. Make a branch for a specific fix/feature ([Git book chapter on branching](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging))
2. Make your commits there. 
3. Push the branch to the forked repo
4. Make a PR

This way you can make several PR's based on branches, each of which has it's own set of commits. :smile:  This [article](https://help.github.com/articles/creating-a-pull-request/) should help!

**Comment by dioptre** [comment]
Thanks for the heads up!

Just out of interest, how would you do a hidden div using elem?

Ie
<div style="display:none;"> is there a way using the syntax you can edit the attributes of tags too?

**Comment by rohitvarkey** [comment]
> Just out of interest, how would you do a hidden div using elem?

`Elem(:div, style = [:visibility=>:hidden])` should do it.

> is there a way using the syntax you can edit the attributes of tags too?

`Elem(:div, attributes => [:a=>2, :b=>3])`

You can check the [Patchwork](https://github.com/shashi/Patchwork.jl) docs for more info.

**Comment by dioptre** [comment]
Great, thanks! 

That looks like a nice way of updating too.

**Comment by shashi** [comment]
Does adding the LaTeX fonts actually fix LaTeX rendering on Linux? I have tried this before and it hadn't worked.

**Comment by dioptre** [comment]
This worked for me on Ubuntu

**Comment by randyzwitch** [comment]
We're still at an impasse here, and the PR has devolved into a discussion.

@dioptre, perhaps you should close this PR, since Shashi has some concerns about code being obsoleted by other WIP and the use of "naked" vs other permutations.

So maybe close this, push the font and latex updates as one PR, the JSON as a second PR and open an issue as a placeholder for Shashi and/or yourself to resolve the "naked" syntax.

**Comment by randyzwitch** [comment]
@dioptre, are you still interested in this feature? If so, please submit it as multiple PRs.

**Pull request closed without merging**